indico/modules/events/contributions/client/js/PublicationButton.jsx

```diff
@@ -0,0 +1,62 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2021 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import publicationURL from 'indico-url:contributions.manage_publication';
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+
+import {Translate} from 'indico/react/i18n';
+import {handleAxiosError, indicoAxios} from 'indico/utils/axios';
+
+import PublicationModal from './PublicationModal';
+
+export default function PublicationButton({eventId, onSuccess}) {
+  const url = publicationURL({event_id: eventId});
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const onPublish = async () => {
+    try {
+      await indicoAxios.put(url);
+      onSuccess();
```

Comment:
I'd move this after the try..catch (and add a `return` in the catch) - that way we don't catch errors in that callback.